lib/pull_preview/license_create.rb

```diff
@@ -0,0 +1,61 @@
+require "slop"
+require "date"
+require "openssl"
+require_relative "license"
+
+module Slop
+  class DateOption < Option
+    def call(value)
+      Date.parse(value)
+    end
+  end
+end
+
+module PullPreview
+  class LicenseCreate
+    def self.run(cli_args)
+      opts = Slop.parse do |o|
+        o.banner = "Usage: pullpreview license-create [options]"
+        o.string "--output-file", "output file"
+        o.string "--type",       "license type [trial]", required: true
+        o.date   "--expires-at", "expiration date"
+        o.int    "--expires-in", "expiration days from today"
+        o.on "--help" do
+          puts command
+          puts
+          puts "To generate a private key, use:"
+          puts "openssl openssl genrsa 2048 -out license_key"
+          puts
+          puts "To generate the matching public key, use:"
+          puts "openssl rsa -in license_key -pubout -out license_key.pub"
+          exit
+        end
+      end
+
+      if opts["expires_in"]
+        opts["expires_at"] = Date.today + opts["expires_in"]
+      end
+
+      # In the license generation application, load the private key from a file.
+      private_key = OpenSSL::PKey::RSA.new File.read("license_key")
```

Comment:
Add a `PullPreview.root_dir` method, that returns a `Pathname` to the root folder (using `__dir__`).

Reply:
Then you can do
```suggestion
      private_key = OpenSSL::PKey::RSA.new PullPreview.root_dir.join("license_key").read
```